### ISSUE TITLE
Potential alternate method to disable MaxxAudioPro on official Dell drivers

### ISSUE THREAD

**Issue by navhaxs** [kevinshroff/KSMRD-Modded-Realtek-Audio-Drivers#17]
I think I've found a method to disable MaxxAudioPro whilst using Dell's official drivers.

"FX Configurator" from this package called "APO Driver" (Link: https://puresoftapps.blogspot.com/2018/04/realtek-apo-driver.html") is a tool which allows the re-configuration of sound driver APOs. It works by modifying the current audio driver installation (probably registry keys?). This thing is intended to be used for sideloading unofficial APOs (I am not recommending nor endorsing this), but for us Dell users we can use it to get rid of the problematic MaxxAudioPro which Dell has forced upon us.

I own a 9570. I tried using the tool on the very latest official Dell audio driver `Realtek-High-Definition-Audio-Driver_FMM28_WIN_6.0.1.8642_A05`.

**Method**:
1. Run *FX Configurator*
2. Click `Product Config Tool`
3. I tried applying `Realtek System Effects - UAD`
- I can't tell the difference between `Realtek System Effects - UAD` and `Realtek System Effects - HDA`, but I think they effectively do the same job as all the GUIDs related to MaxxAudio (all of them??) get removed.
4. Wait for Windows Audio Service to restart, and test audio. Changes are persistent and permanent unless the audio driver is updated.
6. Optional? Install the Realtek Audio Console from https://github.com/alanfox2000/realtek-universal-audio-driver **which lets you configure headset/earphone microphone input via audio jack.** (direct install from Store not working for me).

Audio seems better now, IMO about the same the KSMRD audio driver. i.e. I still get an occasional stutter with either methods.

**My theory**:
So, perhaps all that is needed to disable MaxxAudio could just be a registry patch, to remove the MaxxAudioPro GUID from the APO list... or whatever this tool does? I'm not sure what else the KSMRD driver edits are exactly, but to me the result is the same.

ps. There is a "Waves Audio Effects Component" in Device Manager. Anyone know if this does actually does anything - as disabling it appears to have no effect?

Hope this helps other Dell laptop users out there. You'd expect sound to 'just work' for a $2000+ laptop in 2018.

**Comment by moriel5** [comment]
Actually, all it did was switch over to the UAD model.

There is no difference between that and manually installing the generic UAD drivers, aside from installer weight (Dell's installer is in the hundred's of megabytes, due to the legacy APO interface, complete with MaxxAudio, while downloading the drivers from the unofficial GitHub repo will only weigh ~20-30 megabytes.

**Comment by FranciscoVictor** [comment]
Hi @navhaxs, when you installed the Realtek Audio Console from the _alanfox2000/realtek-universal-audio-driver_, did you install only the Realtek Audio Control (_RtkUWP_w.x.yyy.z_x64_bundle.appxupload_Windows10_PreinstallKit.zip_)?

I'm just checking because I've followed your instructions and additionally also installed the Realtek Audio Console, but the app doesn't work (says that cannot connect to RPC service).

**Comment by navhaxs** [comment]
Yup I used `RtkUWP_1.3.179.0_x64_bundle_ReleaseSign.appxupload_Windows10_PreinstallKit.zip`

I am mostly sure that the UWP app should work as long as the `Realtek Audio Universal Service` is listed in Device Manager. Try restarting just to be sure?

![image](https://user-images.githubusercontent.com/5162718/58416093-d6121c80-80c3-11e9-8d4a-69e07649e953.png)

**Comment by moriel5** [comment]
Based upon your picture, you did not disable MaxxAudio, only had the UWP version of MaxxAudio installed (Waves audio is the same as MaxxAudio).

**Comment by navhaxs** [comment]
@moriel5 after doing the steps I outlined, my MaxxAudioPro app now shows that the device is "not supported", so I reckon it is disabled (app doesn't let you toggle ON the effects) 👍 

![image](https://user-images.githubusercontent.com/5162718/58460751-999cfa00-8171-11e9-8eca-37e383810964.png)

Like I mentioned in the OP, I don't know exactly what `Waves Audio Effects Component` is but it is probably just something like a stub for the installation of the actual files used by the APO.

(I don't see a difference between the UWP and Win32 releases of the MaxxAudioPro app. I'd say that the installation of the app is independent of the APO/audio effects configuration)

**Comment by moriel5** [comment]
Interesting, but I still find it hard to believe that the "enhancements" are not there.

My recommendation, just install the generic UWP Realtek drivers (after removing the Waves/Maxxaudio parts from it), and you can be sure that it will be clean.
Plus, the generic installation files are much lighter, so no point in installing the ones provided by Dell.

**Comment by chasewinnings391** [comment]
I can confirm that this solution works! I'm using the HDA drivers, but using the same steps as described disabled the post-processing of Waves audio!
